libraries/botbuilder-adapter-twilio-whatsapp/src/twilioWhatsAppAdapter.ts

```diff
@@ -57,24 +94,51 @@ export class TwilioWhatsAppAdapter extends BotAdapter {
     protected readonly client: Twilio.Twilio;
     protected readonly channel: string = 'whatsapp';
 
+    // OAuth Properties
+    protected readonly credentials: AppCredentials;
+    protected readonly credentialsProvider: SimpleCredentialProvider;
     /**
      * Creates a new TwilioWhatsAppAdapter instance.
      * @param settings configuration settings for the adapter.
      */
-    public constructor(settings: TwilioWhatsAppAdapterSettings) {
+    public constructor(settings?: Partial<TwilioWhatsAppAdapterSettings>) {
         super();
 
-        this.settings = settings;
+        this.settings = {
+            appId: "",
+            appPassword: "",
+            accountSid: "",
+            authToken: "",
+            phoneNumber: "",
+            endpointUrl: "",
+            ...settings
+        };
+
+        this.credentials = new MicrosoftAppCredentials(
+            this.settings.appId,
+            this.settings.appPassword || "",
+            this.settings.channelAuthTenant
+        );
+
+        this.credentialsProvider = new SimpleCredentialProvider(
+            this.credentials.appId,
+            this.settings.appPassword || ""
+        );
 
         // Add required `whatsapp:` prefix if not exists
-        if (!this.settings.phoneNumber.startsWith('whatsapp:')) {
-            this.settings.phoneNumber = 'whatsapp:' + this.settings.phoneNumber;
+        if (!this.settings.phoneNumber.startsWith("whatsapp:")) {
```

Comment:
Could you run `npm eslint-fix` in the root? We are using some specific style guides and looks like your style settings have changed them.

Reply:
Sure! I didn't know how to run your linter. I will do it as soon as I can.